test/e2e/egress_test.go

```diff
@@ -489,9 +489,12 @@ func testEgressCRUD(t *testing.T, data *TestData) {
 				assert.True(t, exists, "Didn't find desired IP on Node")
 				// Testing the events recorded during creation of an Egress resource.
 				expectedMessage := fmt.Sprintf("Assigned Egress %s with IP %s on Node %v", egress.Name, tt.expectedEgressIP, egress.Status.EgressNode)
-				events, err := data.clientset.CoreV1().Events("").Search(scheme.Scheme, egress)
-				require.NoError(t, err)
-				assert.Contains(t, events.Items[0].Message, expectedMessage)
+				assert.EventuallyWithT(t, func(c *assert.CollectT) {
+					events, err := data.clientset.CoreV1().Events("").Search(scheme.Scheme, egress)
+					if assert.NoError(c, err) && assert.Len(c, events.Items, 1) {
+						assert.Contains(c, events.Items[0].Message, expectedMessage)
+					}
+				}, 2*time.Second, 200*time.Millisecond)
```

Comment:
Not sure what's the default message when assert.EventuallyWithT is timeout. Maybe add a message to get the failure reason quicker: assert.EventuallyWithT(..., "Expected event is not found in time")

Reply:
I used `CollectT` to collect the error. That's the difference between EventuallyWithT and Eventually. So we don't need extra message to indicate which step it fails. An example:
```
=== RUN   TestEgress/testEgressCRUD/single_matching_Node_with_IPv6_range
    egress_test.go:492: 
        	Error Trace:	/var/lib/jenkins/workspace/antrea-ipv6-only-e2e-for-pull-request/test/e2e/egress_test.go:494
        	            				/var/lib/jenkins/golang-releases/go/src/runtime/asm_amd64.s:1650
        	Error:      	"[]" should have 1 item(s), but has 0
    egress_test.go:492: 
        	Error Trace:	/var/lib/jenkins/workspace/antrea-ipv6-only-e2e-for-pull-request/test/e2e/egress_test.go:492
        	Error:      	Condition never satisfied
        	Test:       	TestEgress/testEgressCRUD/single_matching_Node_with_IPv6_range
```